src/connectors/refinement-list/connectRefinementList.js

```diff
@@ -155,137 +155,93 @@ export default function connectRefinementList(renderFn, unmountFn = noop) {
-
-      const canToggleShowMore = canShowLess || canShowMore;
-
-      renderFn(
-        {
-          createURL: _createURL,
-          items,
-          refine,
-          searchForItems: searchFacetValues,
-          instantSearchInstance,
-          isFromSearch,
-          canRefine: isFromSearch || items.length > 0,
-          widgetParams,
-          isShowingMore,
-          canToggleShowMore,
-          toggleShowMore,
-          hasExhaustiveItems,
-        },
-        isFirstSearch
-      );
-    };
-
-    /* eslint-disable max-params */
-    const createSearchForFacetValues = (helper, toggleShowMore) => (
-      state,
-      createURL,
-      helperSpecializedSearchFacetValues,
-      toggleRefinement,
-      instantSearchInstance,
-      isShowingMore
-    ) => query => {
-      if (query === '' && lastResultsFromMainSearch) {
-        // render with previous data from the helper.
-        render({
-          items: lastResultsFromMainSearch,
-          state,
-          createURL,
-          helperSpecializedSearchFacetValues,
-          refine: toggleRefinement,
-          isFromSearch: false,
-          isFirstSearch: false,
-          instantSearchInstance,
-          toggleShowMore, // and yet it will be
-          isShowingMore, // so we need to restore in the state of show more as well
-        });
-      } else {
-        const tags = {
-          highlightPreTag: escapeFacetValues
-            ? TAG_PLACEHOLDER.highlightPreTag
-            : TAG_REPLACEMENT.highlightPreTag,
-          highlightPostTag: escapeFacetValues
-            ? TAG_PLACEHOLDER.highlightPostTag
-            : TAG_REPLACEMENT.highlightPostTag,
-        };
-
-        helper
-          .searchForFacetValues(
-            attribute,
-            query,
-            // We cap the `maxFacetHits` value to 100 because the Algolia API
-            // doesn't support a greater number.
-            // See https://www.algolia.com/doc/api-reference/api-parameters/maxFacetHits/
-            Math.min(getLimit(isShowingMore), 100),
-            tags
-          )
-          .then(results => {
-            const facetValues = escapeFacetValues
-              ? escapeFacets(results.facetHits)
-              : results.facetHits;
-
-            const normalizedFacetValues = transformItems(
-              facetValues.map(({ value, ...item }) => ({
-                ...item,
-                value,
-                label: value,
-              }))
-            );
-
-            render({
-              items: normalizedFacetValues,
-              state,
-              createURL,
-              helperSpecializedSearchFacetValues,
-              refine: toggleRefinement,
-              isFromSearch: true,
-              isFirstSearch: false,
+    let toggleShowMore;
+
+    const createSearchForFacetValues = function(helper) {
+      return (
+        state,
+        createURL,
+        instantSearchInstance,
+        isShowingMore
```

Comment:
would maybe be simpler if these arguments were `renderOptions` so we can forward easily to getWidgetRenderState